### DISCUSSION TITLE
Multiple packages using Prisma in monorepo setup

### DISCUSSION THREAD

**Discussion by lukhol** [prisma/prisma#19444] [Q&A]
I have a monorepo setup in which I have two main directories:
- `apps` - with applications, mostly express based backend-apis
- `packages` - packages with domain logic, most often each of them has it's own database which is using Prisma

Each package can have its own Prisma configuration with its own `schma.prisma`. Then single app (express-based backend API) is using one or more packages. Everything works fine when single app is using only one package with Prisma because I can copy `schema.prisma` location into final `dist` location. Problem is with apps that are using multiple packages and each of them use it's own `schema.prisma`. The question is if there is any way to inline `schema.prisma` into the generated client or pass it to the client during its creation?

**Comment by ludralph** [comment]
Hi @lukhol 👋 

Can you try this workaround as documented in this [issue](https://github.com/prisma/prisma/issues/17687#issuecomment-1437647553)? That user also uses Prisma in a monorepo setup with multiple packages. Let me know if it works for you

**Reply by lukhol** [reply]
Hi, 

thank you for the response! Unfortunately this one will not work for me because I need to bundle my code using webpack. After bundling I have single `server.js` file with all of the Prisma clients bundled into that server.js file. Because of that all of those clients are searching for schema.prisma in the same location.

**Reply by nurul3101** [reply]
Can you maintain a separate Prisma client for each of your packages? This way, each package can have its own database and hence its own Prisma client. To avoid conflicts, you can make sure to put the generated Prisma client of each package in a different output folder. However, this might lead to some redundancy in your code and increased complexity in managing individual Prisma clients.

**Reply by nurul3101** [reply]
We are closing this Discussion due to inactivity. Please open a new Discussion if you have a similar problem 🙏
